data/armies/beasts_of_chaos.json

```
@@ -175,45 +175,103 @@
-      "move": "",
-      "save": "",
-      "bravery": "",
-      "wounds": "",
-      "missile_weapon": [
+      "size": "90 x 52 mm",
+      "move": "See Damage Table",
+      "save": "5+",
+      "bravery": "6",
+      "wounds": "12",
+      "melee_weapon": [
         {
-          "name": "",
-          "range": "",
-          "attacks": "",
-          "to_hit": "",
-          "to_wound": "",
-          "rend": "",
-          "damage": ""
+          "name": "Vicious 'Eadbutt'",
+          "range": "1",
+          "attacks": "1",
+          "to_hit": "4+",
+          "to_wound": "3+",
+          "rend": "-3",
+          "damage": "See Damage Table"
+        },
+        {
+          "name": "Massive Club",
+          "range": "3",
+          "attacks": "See Damage Table",
+          "to_hit": "3+",
+          "to_wound": "3+",
+          "rend": "-1",
+          "damage": "1"
+        },
+        {
+          "name": "Mighty Kick",
+          "range": "2",
+          "attacks": "1",
+          "to_hit": "3+",
+          "to_wound": "3+",
+          "rend": "-2",
+          "damage": "D3"
         }
       ],
-      "melee_weapon": [
+      "abilites": [
         {
-          "name": "",
-          "range": "",
-          "attacks": "",
-          "to_hit": "",
-          "to_wound": "",
-          "rend": "",
-          "damage": ""
+          "name": "Drunken Stagger",
+          "desc": "If a charge roll for this model is a double, this model cannot make a charge move that phase. In addition, the players must roll off. The player who wins the roll-off picks a point on the battlefield 3\" from this model. Each unit within 2\" of that point suffers D3 mortal wounds."
+        },
+        {
+          "name": "Stuff'Em In Me Bag",
+          "desc": "After this model piles in, you can pick 1 enemy model within 3\" of this model and roll a dice. If the roll is equal to or greater than double that model’s Wounds characteristic, it is slain."
+        },
+        {
+          "name": "Whipped into a Frenzy",
+          "desc": "At the start of the combat phase, if this model is within 3\" of any friendly Beasts of Chaos Heroes, you can whip it into a frenzy. If you do so, this model suffers 1 mortal wound, but you can add 1 to the Attacks characteristic of this model’s melee weapons until the end of that phase."
+        },
+        {
+          "name": "Timber!",
+          "desc": "If this model is slain, before removing the model from the battlefield the players must roll off. The player who wins the roll-off picks a point on the battlefield 3\" from this model. Each unit within 2\" of that point suffers D3 mortal wounds. This model is then removed from the battlefield."
         }
       ],
-      "abilites": [
+      "damage_table": [
         {
-          "name": "",
-          "desc": ""
+          "wound_track_position": "1",
+          "move": "8",
+          "massive_club": "3D6",
+          "vicious_'eadbutt": "6",
```

Is there a way we can make these more agnostic? This would make it much easier to statically type on the API side. Something like the following?

This would allow me to keep the objects in the damage table 

```suggestion
		"attacks": [
			 { 
				"name": "Vicious 'eadbutt", 
				"value": "6" 
			 },
		 ]
```

---

Not really, no. Some damage tables have multiple weapons where the attacks are changing, and what would you put when its an ability that changing as the unit gets damaged?

---

> When its an ability that changing as the unit gets damaged?

I'd need to see an example of that one. 

But either way we would need to find a way to make the key's neutral, otherwise this would require dozens and dozens of new types and become a massive challenge to maintain a universal schema.

---

Here's another example of how to structure this into columns with agnostic keys.

```
{
  "damage_table": [
    {
      "name": "Wounds Suffered",
      "values": [
        "0-2",
        "3-4",
        "5-7",
        "8-9",
        "10+"
      ]
    },
    {
      "name": "Massive Impaling Talons",
      "values": [
        "2+",
        "2+",
        "3+",
        "3+",
        "4+"
      ]
    },
    {
      "name": "Sweeping Blows",
      "values": [
        "3",
        "2",
        "2",
        "1",
        "1"
      ]
    },
    {
      "name": "Doom Tendril Staff",
      "values": [
        "2+",
        "3+",
        "4+",
        "5+",
        "6+"
      ]
    }
  ]
}
```